### PR TITLE
nixos/oci-containers: Remove dependency on system.path

### PR DESCRIPTION
**Pull request by szlend**
###### Motivation for this change

Every oci-containers service depends on `system.path` which means it will be needlessly restarted every time a package is added/removed from the system path. To fix this I just launch the container backend through bash. Since the backend is already configured through the service `PATH`, it will automatically pick the correct one.

I also noticed that we weren't taking into account the configured `docker.package` when using the docker backend, so I also fixed that.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/virtualisation/oci-containers.nix [hunk 1]
```diff
@@ -233,7 +233,8 @@ let
       StandardOutput = "null";
       StandardError = "null";
       ExecStart = concatStringsSep " \\\n  " ([
-        "${config.system.path}/bin/${cfg.backend} run"
+        ''${pkgs.bash}/bin/sh -c "''
```
Comment (Infinisil) [inline nixos/modules/virtualisation/oci-containers.nix:236]: I suggest using
```nix
{
  systemd.serivces.*.script = ''
    exec ...
  '';
}
```

instead

Comment (szlend) [reply nixos/modules/virtualisation/oci-containers.nix:236]: Good call

**Comment by szlend** [comment]
@Infinisil I replaced `ExecStart` and `ExecStop` with `script` and `preStop`. I don't know how to replace `concatStringsSep` with a string literal though. I think the way it's done currently works very well with optionals and empty lists without leaving gaps in the generated script. But if you have a good proposal I'd be happy to do it.

**Comment by adisbladis** [comment]
@GrahamcOfBorg test oci-containers

**Comment by adisbladis** [comment]
@GrahamcOfBorg test oci-containers

**Comment by adisbladis** [comment]
Thank you!

**Pull request merged**